scripts/webrtcclientsdk.js

```diff
@@ -124,6 +126,20 @@ define(function (require) {
         //Uncomment the below line, if we want to change device in-meeting
         MediaStarted = true;
 
+        if (initializing) {
```

Comment:
This is the fix. This block of code is executed _after_ `RTCManager.getLocalMedia` has been fulfilled. It may help to view the expanded file, so you can see how it's executed within the `RTCManager.getLocalMedia` success callback. (See [here](https://github.com/mirainc/sdk-webrtc-meetings/blob/sebko/feat/premute/scripts/webrtcclientsdk.js#L129-L141))

Reply:
Do we need to fork the sdk to fix this? We should really only do this unless we need to because now updating to the latest is going to be that much more work.

Couldn't we call the mute function in the embed sdk right after initializing here? https://github.com/mirainc/bluejeans-meeting-embed/blob/master/src/initialize.js#L42

Reply:
I've tried calling the mute function right after initializing, and the call is ignored, resulting in the UI incorrectly indicating a muted mic. (See here: https://raydiant.slack.com/archives/C0127EB8T9U/p1600299868013900)

I sent the example to Andrew, but have not heard back from him yet. I mentioned this to David and his recommendation was to fork and fix the issue ourselves.

Reply:
I agree that we should only fork after all other avenues have been exhausted. I'm afraid I'm not sure how to fix this issue without touching their SDK.

Reply:
I can get slack web links to work for some reason but forking should really be a last resort. 

I tried my best not to change the SDK in the bjn embed so it would be relatively painless to update. Now we will need to re-implement this code. I've been here before and it sucks, they could completely change the code base making a diff impossible. If we _need_ to fork we should change the absolute minimum.

In any case from what I can tell the only difference between this `initialize` block here and calling `toggleAudioMute` after initialization is that here the call is run after the getLocalMedia promise. 

Does a setTimeout work? A hacky fix in our code is better than forking theirs. 

If setTimeout doesn't work it seems like it would be a pretty minor change to fork the sdk to return the promise from `initialize`. I could see having a better change of getting this merged in upstream into the actual SDK, which would the ideal goal of working

Reply:
>Does a setTimeout work?
We could use a timeout, but it's hard to estimate how long it takes for the getLocalMedia call to fulfill.
What if 95% of the time it takes 1 second, but for some devices it takes 5 seconds or longer? I guess we could set it to a really high number.

In the BJN embed, we're calling the mute function after joining the call - that gives us a few seconds for the client to initializes and, more importantly, fulfill the `getLocalMedia` promise. So we're effectively using a 1-2 second timeout here, but David has mentioned he has experienced the "UI incorrectly shows muted" bug before.

Reply:
> fork the sdk to return the promise from initialize
I can look into this. The other alternative is to provide an `onLocalMediaReady` callback to the initialize call.

Reply:
> We could use a timeout, but it's hard to estimate how long it takes for the getLocalMedia call to fulfill.
What if 95% of the time it takes 1 second, but for some devices it takes 5 seconds or longer? I guess we could set it to a really high number.

Is that what happens or does `setTimeout(() => {}, 0 - 100)` just work? If that doesn't what if you await your all getUserMedia call before calling mute to get the timing right? Or is there some internal state you can poll to see when you can call mute? None of these options would be any worse than what I'm already doing in the Zoom SDK.

If there's no workaround that doesn't involve editing the SDK then the goal should be to make a feature request to BlueJeans with an actual PR that could potentially be merged. The PR is more likely to be merged if it contains the minimum amount of changes necessary to make a fix. At the very least you can point Andrew to the PR and ask him to think of a better approach.

Reply:
>does setTimeout(() => {}, 0 - 100) just work?

No, it's not working for me. Feel free to give it a shot on your machine. https://github.com/mirainc/sdk-webrtc-meetings/pull/1/files#diff-65d7f35d2dc8f88273f9b1e3438d3af2R108 (Just update 3000ms to 100ms)

My concern is that under certain conditions, It can take a while for the `getLocalMedia` call to fulfil and it's hard to estimate that upper limit.

>what if you await your own getUserMedia call before calling mute to get the timing right

Good idea. We could even await BlueJeans's `RTCManager.getLocalMedia` call from the BJN embed.